tests/roles/backend_services/tasks/main.yaml

```diff
@@ -32,116 +32,43 @@
         oc set data secret/osp-secret "PlacementPassword={{ placement_password }}"
     {% endif %}
 
-- name: deploy backend services
+- name: Gather min facts if we dont have ansible_user_dir
+  ansible.builtin.setup:
+    gather_subset: "!min,user_dir"
+  when: ansible_user_dir is not defined
+
+- name: when not periodic, just run kustomize on the base/deployment
+  when: not periodic|default(false)
   ansible.builtin.shell: |
     {{ shell_header }}
     {{ oc_header }}
-    oc apply -f - <<EOF
-    apiVersion: core.openstack.org/v1beta1
-    kind: OpenStackControlPlane
-    metadata:
-      name: openstack
-    spec:
-      secret: osp-secret
-      storageClass: local-storage
-
-      cinder:
-        enabled: false
-        template:
-          cinderAPI: {}
-          cinderScheduler: {}
-          cinderBackup: {}
-          cinderVolumes: {}
-
-      dns:
-        enabled: true
-        template:
-          externalEndpoints:
-          - ipAddressPool: ctlplane
-            loadBalancerIPs:
-            - 192.168.122.80
-          options:
-          - key: server
-            values:
-            - 192.168.122.1
-          replicas: 1
-
-      glance:
-        enabled: false
-        template:
-          glanceAPIInternal: {}
-          glanceAPIExternal: {}
-
-      horizon:
-        enabled: false
-        template: {}
-
-      ironic:
-        enabled: false
-        template:
-          ironicConductors: []
-
-      keystone:
-        enabled: false
-        template: {}
-
-      manila:
-        enabled: false
-        template:
-          manilaAPI: {}
-          manilaScheduler: {}
-          manilaShares: {}
-
-      mariadb:
-        templates:
-          openstack:
-            storageRequest: 500M
-
-      memcached:
-        enabled: true
-
-      neutron:
-        enabled: false
-        template: {}
-
-      nova:
-        enabled: false
-        template: {}
-
-      ovn:
-        enabled: false
-        template:
-          ovnController:
-            external-ids:
-              system-id: "random"
-              ovn-bridge: "br-int"
-              ovn-encap-type: "geneve"
-
-      placement:
-        enabled: false
-        template: {}
-
-      rabbitmq:
-        templates:
-          rabbitmq:
-            externalEndpoint:
-              loadBalancerIPs:
-              - 172.17.0.85
-              ipAddressPool: internalapi
-              sharedIP: false
-            replicas: 1
-          rabbitmq-cell1:
-            externalEndpoint:
-              loadBalancerIPs:
-              - 172.17.0.86
-              ipAddressPool: internalapi
-              sharedIP: false
-            replicas: 1
-
-      telemetry:
-        enabled: false
-        template: {}
-    EOF
+    kustomize build base > {{ ansible_user_dir }}/test_deployment.yaml
```

Comment:
As discussed on the call, let's not write anything into the user home directory. E.g. i run this thing on my laptop :).

Reply:
@marios I think when executing an Ansible playbook, the working dir of the ansible-plabook process is the directory where the playbook resides. We already use a tmp directory relative to that for MariaDB [1] and OVN [2] and it is in gitignore [3]. We can perhaps write the deployment yaml there.

[1] https://github.com/openstack-k8s-operators/data-plane-adoption/blob/3232403e6baef84ec267593c98745cc01dfd9093/tests/roles/mariadb_copy/defaults/main.yaml
[2] https://github.com/openstack-k8s-operators/data-plane-adoption/blob/3232403e6baef84ec267593c98745cc01dfd9093/tests/roles/ovn_adoption/defaults/main.yaml
[3] https://github.com/openstack-k8s-operators/data-plane-adoption/blob/3232403e6baef84ec267593c98745cc01dfd9093/.gitignore#L4